src/main/com/smithsmodding/armory/common/crafting/blacksmiths/recipe/VanillaAnvilRecipe.java

```diff
@@ -183,7 +172,8 @@ private void calculateValues() {
             if (iRightInputStack != null) {
                 if (!checkForge(iLeftInputStack, iRightInputStack, iEntity.getState().getItemName(), tBaseLevelExperienceCost))
                     return;
-                tFlagIsEnchantableByBook = iRightInputStack.getItem() == Items.ENCHANTED_BOOK && Items.ENCHANTED_BOOK.getEnchantments(iRightInputStack).tagCount() > 0;
+
+                tFlagIsEnchantableByBook = iRightInputStack.getItem() == Items.ENCHANTED_BOOK && ItemEnchantedBook.getEnchantments(iRightInputStack).tagCount() > 0;
 
                 if (tEventStack.isItemStackDamageable() && tEventStack.getItem().getIsRepairable(iLeftInputStack, iRightInputStack)) {
                     tRepairAmount = Math.min(tEventStack.getItem().getDamage(tEventStack), tEventStack.getMaxDamage() / 4);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 4 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)